src/quests.h

```diff
@@ -30,19 +41,35 @@ class Mission
 	std::string getDescription(Player* player) const;
 	uint16_t getID() const { return id; }
 	uint32_t getStorageId() const { return storageID; }
+	void setStorageId(uint32_t storageId) { storageID = storageId; }
 	int32_t getStartStorageValue() const { return startValue; }
+	void setStartStorageValue(int32_t startStorageValue) { startValue = startStorageValue; }
 	int32_t getEndStorageValue() const { return endValue; }
+	void setEndStorageValue(int32_t endStorageValue) { endValue = endStorageValue; }
+	bool getIgnoreEndValue() const { return ignoreEndValue; }
+	void setIgnoreEndValue(bool ignoreEndValue) { ignoreEndValue = ignoreEndValue; }
 
 	std::map<int32_t, std::string> descriptions;
 	std::string mainDescription;
 
+	MissionCallback* getCallback()
+	{
+		if (!callback) {
+			callback.reset(new MissionCallback());
+		}
+
+		return callback.get();
+	}
```

Comment:
If this can't be empty, just make the `callback` be a `MissionCallback` and return a reference to it.

Reply:
there are other places in files that are handled this way, and there won't always be a value in callback, only when using a callback.